src/stream/reader/logfile.rs

```diff
@@ -206,13 +206,22 @@ impl LogFile {
                 Ok(())
             };
 
-        if let Some(task_id) = opts.task {
-            if let Some(task_info) = self.index.get(&task_id) {
-                let instance = task_info.last_instance();
-                if !instance.finished && opts.allow_unfinished {
-                    anyhow::bail!("Stream for task {} is not finished", task_id);
+        if let Some(ref array) = opts.task {
```

Comment:
Now that I look at it, both branches are pretty much the same. Could you try to unify the code? Either get all of the indices into an Vec or first load a Vec of tasks and then use it.